### PR TITLE
fix case where wells are shut behind our back

### PR DESCRIPTION
**Pull request by totto82**
Also avoid duplication of to equal methods (getWell and well)

### CODE REVIEW

**Comment by akva2** [comment]
strictly speaking the methods have different behaviors (throw vs assert). not sure what is wanted.

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1645,6 +1629,21 @@ namespace Opm {
...
+    hasWell(const std::string& well_name) const
+    {
+        // finding the iterator of the well in wells_ecl
+        auto well = std::find_if(well_container_.begin(),
+                                     well_container_.end(),
+                                     [&well_name](const WellInterfacePtr& elem)->bool {
+                                         return elem->name() == well_name;
+                                     });
+
+        return (well != well_container_.end());
```
Comment (bska) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:1644]: Using [`std::any_of()`](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of) this is equivalent to
```C++
return std::any_of(well_container_.begin(), well_container_.end(),
    [&well_name](const WellInterfacePtr& elem) -> bool
{
    return elem->name() == well_name;
});
```

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
This should fix the current issue in master with norne 4_MSW. Lets trigger benchmark and check

**Comment by totto82** [comment]
benchmark please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/wells/BlackoilWellModel_impl.hpp:1644]: Looks better, I will update, but wait until benchmark is done to avoid disruption.

**Comment by akva2** [comment]
*cough* https://github.com/OPM/opm-simulators/pull/3429 *cough*

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.998    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.003    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1310

**Comment by totto82** [comment]
benchmark please

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1        |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.01     |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.996    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1310

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 0.999    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.01     |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.003    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.001    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.003    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1311

**Comment by totto82** [comment]
Strange the `4_MSW` works locally. I will re-run and report.

**Comment by totto82** [comment]
`4_MSW` still work locally with this PR. @alfbr Any idea of why benchmark still fails?

**Comment by alfbr** [comment]
I believe it fails because there is no reference benchmark. Being broken in master, there is no run time to compare to, and I believe that is why it is not shown. Unless there are reasons not to merge you fix, I suggest we merge it. We will then hopefully have complete reference runs tomorrow.

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1        |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 1.01     |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 1.004    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1311

**Pull request merged**